lib/generate-expected-shas.ts

```diff
@@ -0,0 +1,40 @@
+import yargs from 'yargs';
+import { readFileSync, writeFileSync } from 'fs';
+import { join } from 'path';
+import { log } from './log';
+import { version } from '../package.json';
+
+async function main() {
+  const { argv } = yargs
+    .option('input', { alias: 'i', default: 'shas.txt', type: 'string' })
+    .version(version)
+    .alias('v', 'version')
+    .help()
+    .alias('h', 'help');
+
+  const shaFileContent = readFileSync(argv.input).toString();
+  const shaMap: { [id: string]: string } = {};
+  for (const line of shaFileContent.split('\n')) {
+    const lineParts = line.split(/\s+/);
+    if (lineParts.length === 2) {
+      // eslint-disable-next-line prefer-destructuring
+      shaMap[lineParts[1]] = lineParts[0];
+    }
+  }
+
+  // Sort map
+  const sortedShaMap: { [id: string]: string } = {};
+  for (const nodeVersion of Object.keys(shaMap).sort()) {
```

Comment:
yes, probably

Reply:
added a semver based sorting approach, using semver.coerce to force the full node binary filenames to valid semantic versions, this let me avoid using any regex to extract the version portion.

The risk here is that I think now the sorting within a version depends on the original sorting of the input file, which may depend on the OS file listing sorting when we generate the shas.txt file. I think this may result in one more commit later on if the sorting within each version isn't the same as this PR's sorting.